### ISSUE TITLE
Dangling repository links in certain cases

### ISSUE THREAD

**Issue by dberzano** [alisw/alibuild#88]
Test case: **ThePEG** depends on **pythia**.

We first build **pythia** on a certain machine and we decide not to publish the build results (*i.e.* no `::rw` option).

If we then build **ThePEG**, this time using the `::rw` option, and the build ends on the same machine, aliBuild finds the local copy of the **pythia** build and it does not rebuild it, therefore it does not upload the package.

Instead, inside the **ThePEG** `dist*` directories, it creates dangling links to a non-existing repository version of **pythia**.

Example builds exhibiting the problem: [pythia](https://alijenkins.cern.ch/job/build-any-ib/2143/), [ThePEG](https://alijenkins.cern.ch/job/build-any-ib/2144/).

**Comment by ktf** [comment]
I agree this is a bug. For the moment just make sure you always build and upload the toplevel package you want to use, not an intermediate one.
